Description: WinFF adds extra zeros to time parts every time a conversion
  is started, producing arguments like -ss 0:00002:00000 (causing ffmpeg to
  fail). This patch prevents the argument from growing.
Author: Matthew Weatherford <matt@biggmatt.com>
Origin: upstream, http://code.google.com/p/winff/source/detail?r=523
Bug: http://code.google.com/p/winff/issues/detail?id=111
Reviewed-by: Paul Gevers <paul@climbing.nl>

Index: winff-git/unit1.pas
===================================================================
--- winff-git.orig/unit1.pas	2011-03-18 18:11:40.372911427 +0100
+++ winff-git/unit1.pas	2011-03-18 18:14:43.256911429 +0100
@@ -1482,8 +1482,8 @@
    if edtSeekHH.Value + edtSeekMM.Value + edtSeekSS.Value > 0 then
    begin
      ignorepreview := true;
-     if edtSeekMM.Value < 10 then edtSeekMM.Text := '0' + edtSeekMM.Text;
-     if edtSeekSS.Value < 10 then edtSeekSS.Text := '0' + edtSeekSS.Text;
+     if (edtSeekMM.Value < 10) and (length(edtSeekMM.Text)<2) then edtSeekMM.Text := '0' + edtSeekMM.Text;
+     if (edtSeekSS.Value < 10) and (length(edtSeekSS.Text)<2) then edtSeekSS.Text := '0' + edtSeekSS.Text;
 
      commandline:=replaceparam(commandline,'-ss','-ss ' + edtSeekHH.Text + ':' + edtSeekMM.Text + ':' + edtSeekSS.Text);
    end;
@@ -1491,8 +1491,8 @@
    if edtTTRHH.Value + edtTTRMM.Value + edtTTRSS.Value > 0 then
    begin
      ignorepreview := true;
-     if edtTTRMM.Value < 10 then edtTTRMM.Text := '0' + edtTTRMM.Text;
-     if edtTTRSS.Value < 10 then edtTTRSS.Text := '0' + edtTTRSS.Text;
+     if (edtTTRMM.Value < 10) and (length(edtTTRMM.Text)<2)  then edtTTRMM.Text := '0' + edtTTRMM.Text;
+     if (edtTTRSS.Value < 10) and (length(edtTTRSS.Text)<2)  then edtTTRSS.Text := '0' + edtTTRSS.Text;
 
      commandline:=replaceparam(commandline,'-t','-t ' + edtTTRHH.Text + ':' + edtTTRMM.Text + ':' + edtTTRSS.Text);
    end;
